junebug/amqp.py

```diff
@@ -0,0 +1,115 @@
+from twisted.internet.defer import inlineCallbacks, returnValue, Deferred
+from twisted.internet.protocol import ReconnectingClientFactory
+from twisted.python import log
+from txamqp.client import TwistedDelegate
+from txamqp.content import Content
+from txamqp.protocol import AMQClient
+from vumi.utils import vumi_resource_path
+from vumi.service import get_spec
+
+
+class AmqpFactory(ReconnectingClientFactory, object):
```

Comment:
From slack discussion, we are thinking of making a new PR later for changing the in-process reconnecting behaviour to be configurable via a config flag so that one can choose to enable or disable reconnecting (and have the process exit if reconnecting is disabled).